### PR TITLE
[sortablejs] update `onMove()` to 2 arguments & add new properties in `Options`

### PR DESCRIPTION
**Pull request by waynevanson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/SortableJS/Sortable#options>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/sortablejs/index.d.ts [hunk 1]
```diff
@@ -28,6 +29,15 @@ declare class Sortable {
...
     static create(element: HTMLElement, options: Sortable.Options): Sortable;
 
+    /**
+     * Mounts a plugin to the `Sortable`
+     * @param plugins One or more of the sortable plugins.
+     */
+    // todo:
+    // + tests
+    // + declarations for the plugins. Plugins should extend a `Sortable.Plugin` class.
+    static mount(...plugins: any[]): void;
```
Comment (MaartenStaa) [inline types/sortablejs/index.d.ts]: It seems like a missed opportunity to not also add types now for the plugins. This is better than nothing, but still.

File: types/sortablejs/index.d.ts [hunk 2]
```diff
@@ -137,6 +147,15 @@ declare namespace Sortable {
...
         delay?: number;
+        /**
+         * Only delay if user is using touch
+         */
+        delayOnTouchOnly?: boolean;
+        /**
+         * Direction of Sortable
+         * (will be detected automatically if not given)
+         */
+        direction?: 'horizontal' | 'vertical';
```
Comment (MaartenStaa) [inline types/sortablejs/index.d.ts]: This isn't right, `direction` can also be a function. Example in the docs:

```javascript
Sortable.create(el, {
	direction: function(evt, target, dragEl) {
		if (target !== null && target.className.includes('half-column') && dragEl.className.includes('half-column')) {
			return 'horizontal';
		}
		return 'vertical';
	}
});
```

File: types/sortablejs/index.d.ts [hunk 3]
```diff
@@ -255,7 +332,7 @@ declare namespace Sortable {
         /**
          * Event when you move an item in the list or between lists
          */
-        onMove?: (event: MoveEvent) => boolean;
+        onMove?: (evt: MoveEvent, originalEvent: Event) => void;
```
Comment (MaartenStaa) [inline types/sortablejs/index.d.ts]: This return type shouldn't be `void`. The docs say:

```javascript
		// return false; — for cancel
		// return -1; — insert before target
		// return 1; — insert after target
```

Though, reading the actual code, it seems returning `true` would also work. So maybe `=> boolean | -1 | 1`?

**Review by MaartenStaa — CHANGES_REQUESTED** [review]

File: types/sortablejs/index.d.ts (hunk 1 above)
Comment (waynevanson) [reply types/sortablejs/index.d.ts]: It does indeed. I would love to but I'm not too familiar with all the plugins.

I plan to add more to this at a later date in another pull.

File: types/sortablejs/index.d.ts (hunk 2 above)
Comment (waynevanson) [reply types/sortablejs/index.d.ts]: Ok, I will add this in.

File: types/sortablejs/index.d.ts (hunk 3 above)
Comment (waynevanson) [reply types/sortablejs/index.d.ts]: Yes, that is correct.

I will change this. That was a mistake on my part.

**Comment by waynevanson** [comment]
I'm actually going to close this because I just rewrote a lot of new changes, including all the plugins.

I changed all that was requested.

**Pull request closed without merging**